cluster-operator/src/main/java/io/strimzi/operator/cluster/ClusterOperator.java

```diff
@@ -110,11 +112,15 @@ public void start(Future<Void> start) {
                     });
                     return startHealthServer().map((Void) null);
                 })
-                .compose(start::complete, start);
+                .compose(i -> {
+                    ((Promise<Void>) start).complete(i);
```

Comment:
So it's interesting that in the Vert.x 3.8.x they started to migrate to `Promise` but left `Future` here while the master branch has a `Promise` in the start/stop of the `AbstractVerticle`. Why do we need a cast to `Promise` here? I see this method is still using `Future` for everything (see i.e. `watchFutures`)

Reply:
Yeah, it does not make sense to me neither. Unluckily, they use `Future` in 3.8 (https://github.com/eclipse-vertx/vert.x/blob/3.8/src/main/java/io/vertx/core/AbstractVerticle.java#L105)

But in the master (https://github.com/eclipse-vertx/vert.x/blob/master/src/main/java/io/vertx/core/AbstractVerticle.java#L105) they have already a `Promise`, so this casting will be aliminated with next version, I believe.

Reply:
I suppose they can't change `start(Future)` to `start(Promise)` without breaking binary compatibility, which is something they can only do in 4.0.

Reply:
So, do we expect something from Stanislav here? Or can it be reviewed / merged?

Reply:
at this point, nothing I am approving this.